python/cuml/cuml/accel/estimator_proxy.py

```diff
@@ -135,6 +135,11 @@ def __init__(self, *args, **kwargs):
         # be pickled properly.
         cls.__module__ = cls._gpu_class._cpu_class_path.rsplit(".", 1)[0]
 
+        # Forward _estimator_type as a class attribute if available
+        _estimator_type = getattr(cls._cpu_class, "_estimator_type", None)
+        if isinstance(_estimator_type, str):
+            cls._estimator_type = _estimator_type
```

Comment:
I think we should explicitly codify that we expect `_estimator_type` to be either `None` or `str`.

```suggestion
        if _estimator_type := getattr(cls._cpu_class, "_estimator_type", None) is not None:
            assert isinstance(_estimator_type, str)
            cls._estimator_type = _estimator_type
```

Reply:
Theoretically it could be an instance property instead of a class value, which is why I wrote it the way I did (only forwarding if it's a str class attribute).